src/subscription.ts

```diff
@@ -254,14 +254,19 @@ export class Subscription extends (EventEmitter as new () => TypedEventEmitter<S
     if (this._setState(SubscriptionState.Subscribing)) {
       this.emit('subscribing', { channel: this.channel, code: code, reason: reason });
     }
-    this._subscribe(false, false);
+    if (code === subscribingCodes.subscribeCalled) {
```

Comment:
setSubscribing should try to subscribe – this is important to handle async unsubscribe pushes. See https://github.com/centrifugal/centrifuge-js/blob/e8d553bff197d7b06d2be8255cde339b896f2ad0/src/centrifuge.ts#L1676

Another concern here - subscribingCodes are mostly informational, it better to avoid putting logic based on them.

I think it's important to mark transport not open right after disconnect called, at that point no other commands should be sent to server. Or did you find some problems with that?

Probably, to avoid calling `this._subscribe` here and doing extra work it's possible to check `_transportIsOpen` property to be `true` here instead of `code === subscribingCodes.subscribeCalled` check.

Reply:
Oh, looks liked I missed this `_setSubscribing`, that is coming from unsubscribe and thought that there's only 2 codes for calling subscribe. Sure it's better to avoid logic based on codes, that is not fully controlled by SDK

Reverted it and moved _transportIsOpen to the top, looks like the best choice for now

The only problem I see, is not really problem) It's not direct dependence, which increases the scope of required knowledge about inner logic to debug